packages/extension/src/contexts/LedgerContext/LedgerContext.tsx

```diff
@@ -768,6 +875,62 @@ const LedgerProvider: FC = ({ children }) => {
     }
   }, [client, getCurrentWallet]);
 
+  const getNFTInfo = useCallback(
+    async (NFTokenID: string, network?: string): Promise<NFTInfoResponse> => {
+      const connectToLedger = async (server: string): Promise<Client> => {
```

Comment:
I think this function to connect to the ledger could be extracted to its own file. Isn't there already a way to connect to the ledger (in a different file maybe?) ?